src/plugins/navigation_embeddable/public/embeddable/navigation_embeddable_factory.ts

```diff
@@ -64,33 +99,32 @@ export class NavigationEmbeddableFactoryDefinition
       reduxEmbeddablePackage,
       { editable },
       { ...getDefaultNavigationEmbeddableInput(), ...initialInput },
+      getNavigationEmbeddableAttributeService(),
       parent
     );
   }
 
   public async getExplicitInput(
-    initialInput?: NavigationEmbeddableInput,
+    initialInput: NavigationEmbeddableInput,
     parent?: DashboardContainer
-  ) {
+  ): Promise<Omit<NavigationEmbeddableInput, 'id'>> {
     if (!parent) return {};
 
     const { openEditorFlyout } = await import('../editor/open_editor_flyout');
 
     const input = await openEditorFlyout(
-      { ...getDefaultNavigationEmbeddableInput(), ...initialInput },
+      {
+        ...getDefaultNavigationEmbeddableInput(),
+        ...initialInput,
+      },
       parent
-    ).catch(() => {
-      // swallow the promise rejection that happens when the flyout is closed
-      return {};
-    });
```

Comment:
No need to `catch` here as the Embeddable panel will handle the catch and return without creating an embeddable.

Reply:
I think we still need this `catch` because we're not getting an `Unhandled promise rejection` error thrown to the console on cancel:


https://github.com/elastic/kibana/assets/8698078/ee3c018d-ebc2-49bb-90cf-b5d5fa92df02